backend/src/Edna/Upload/Web/Types.hs

```diff
@@ -51,16 +51,29 @@ instance Buildable FileBS where
 instance ToSchema FileBS where
   declareNamedSchema _ = pure $ NamedSchema (Just "ByteString") binarySchema
 
-instance HasSwagger api => HasSwagger (MultipartForm Mem FileBS :> api) where
-  toSwagger _ = toSwagger (Proxy :: Proxy api) & addParam param
+instance HasOpenApi api => HasOpenApi (MultipartForm Mem FileBS :> api) where
+  toOpenApi _ = toOpenApi (Proxy :: Proxy api) & addRequestBody body
     where
-      param = mempty
-        & name .~ "file"
-        & required ?~ True
-        & description ?~ "Experiment to parse"
-        & schema .~ ParamOther (mempty
-            & in_ .~ ParamFormData
-            & paramSchema .~ (mempty & type_ ?~ S.SwaggerFile))
+      body :: O.RequestBody
+      body = mempty
+        & O.content .~ content
+        & O.required ?~ True
+
+      content :: Map.InsOrdHashMap MediaType O.MediaTypeObject
+      content = Map.singleton "multipart/form-data" $ mempty
+        & O.schema ?~ paramSchema
+
+      paramSchema :: O.Referenced O.Schema
+      paramSchema = O.Inline $ mempty
+        & O.type_ ?~ O.OpenApiObject
+        & O.required .~ ["file"]
+        & O.properties .~ props
+
+      props :: Map.InsOrdHashMap Text (O.Referenced O.Schema)
+      props = Map.singleton "file" $ O.Inline $ mempty
```

Comment:
likewise, I'd use `at` and merge props directly into `paramSchema`, w/o creating additional binding, like

```haskell
  & O.properties .~ (mempty
    at "file" ?~ (mempty & ....)
```

Reply:
I was looking for something like this! thanks